3D Model preview

dmitry-yudakov opened the pull request.

dmitry-yudakov committed:
feat(creator): initial version of 3D uploaded model previewer

PAKovalev commented on packages/map3d/src/components/Model3dPreview/Model3dPreview.tsx:
```diff
@@ -0,0 +1,119 @@
+
+import * as styled from './Model3dPreview.styled';
+
+const createScene = (canvas: HTMLCanvasElement) => {
```
It looks like a hook

dmitry-yudakov replied:
It could be but it will all change when we're able to use react-three-fiber

dmitry-yudakov committed:
feat(creator): transparent background

dmitry-yudakov committed:
feat(creator): show preview on hovering item in assets library

dmitry-yudakov committed:
feat(creator): progress bar for preview

dmitry-yudakov committed:
feat(creator): better background for 3D model preview

dmitry-yudakov committed:
fix(creator): better styling for 3D model preview

dmitry-yudakov committed:
Merge remote-tracking branch 'origin/develop-plugins' into feature/3d-model-preview

dmitry-yudakov committed:
fix(creator): cannot render another model if showing already loaded one

dmitry-yudakov committed:
feat(creator): delay before loading the 3d model preview

Useful for onHover behaviour - should prevent too many loads on non-intentional hover

dmitry-yudakov committed:
Merge branch 'develop-plugins' into feature/3d-model-preview

dmitry-yudakov committed:
Merge branch 'develop-plugins' into feature/3d-model-preview

dmitry-yudakov committed:
fix(creator): world breaks after certain number of previews instantiated

It seems the webgl renderer needs to be explicitly disposed

dmitry-yudakov committed:
refactor(creator): different function for detecting hover over grids

dmitry-yudakov committed:
feat(creator): 3d model preview - auto-rotate

dmitry-yudakov committed:
feat(creator): 3d model preview - auto position camera

This centers the object and adjusts the camera position

dmitry-yudakov committed:
feat(creator): 3d model preview - improve lights

dmitry-yudakov committed:
feat(creator): use 3d model preview in spawn object page

FIZLIN reviewed (approved):
LGTM
